app/views/dashboard/_subscription_widget.html.erb

```diff
@@ -5,12 +5,15 @@
 <% unless Organization.current.present? %>
   <p class="ca"><%= _("Please select an organization to view subscription status.") %></p>
 <% else %>
-  <% owner_info = Organization.current.owner_info %>
-  <% invalid_consumer_count = owner_info.total_invalid_compliance_consumers %>
-  <% partial_consumer_count = owner_info.total_partial_compliance_consumers %>
-  <% valid_consumer_count = owner_info.total_valid_compliance_consumers %>
-  <% total_count = owner_info.total_consumers %>
-  <% subscription_status_url = '/content_hosts?search='  + ERB::Util.url_encode('subscription_status = ') %>
+  <%host = Host::Managed.authorized('view_hosts', ::Host::Managed).where(:organization => Organization.current)%>
+  <% total_count = host.size %>
+  <% partial_consumer_count = Host::Managed.authorized('view_hosts', ::Host::Managed).where(:organization => Organization.current).search_for("subscription_status = partial").size %>
+  <% valid_consumer_count = Host::Managed.authorized('view_hosts', ::Host::Managed).where(:organization => Organization.current).search_for("subscription_status = valid").size%>
+  <% invalid_consumer_count = Host::Managed.authorized('view_hosts', ::Host::Managed).where(:organization => Organization.current).search_for("subscription_status = invalid").size%>
+  <% unknown_consumer_count = Host::Managed.authorized('view_hosts', ::Host::Managed).where(:organization => Organization.current).search_for("subscription_status = unknown").size %>
+  <% unregistered_consumer_count = total_count - partial_consumer_count - valid_consumer_count - invalid_consumer_count%>
+  <% subscription_status_url = '/content_hosts?search='%>
+  <% registered_subscription_url = subscription_status_url + ERB::Util.url_encode('subscription_status = ') %>
```

Comment:
@jlsherrill : Moved this part to a helper..You can take a look.
I am writing tests for the same and waiting for the build.